src/App.tsx

```diff
@@ -16,35 +23,72 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [reverse, setReverse] = useState<boolean>(false);
+
+  const visibleGoods = [...goodsFromServer].sort(
+    (good1: string, good2: string) => {
+      switch (sortField) {
+        case SortType.ALPHABETICALLY:
+          return good1.toLowerCase().localeCompare(good2.toLowerCase());
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    },
+  );
+
+  if (reverse) {
+    visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.ALPHABETICALLY)}
+          type="button"
+          className={`button is-info ${sortField !== SortType.ALPHABETICALLY ? 'is-light' : ''}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.LENGTH)}
+          type="button"
+          className={`button is-success ${sortField !== SortType.LENGTH ? 'is-light' : ''}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setReverse(!reverse)}
+          type="button"
+          className={`button is-warning ${!reverse ? 'is-light' : ''}`}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== '' || reverse) && (
```

Comment:
The condition `(sortField !== '' || reverse)` for displaying the `Reset` button is correct, but it might be more readable to use `SortType.DEFAULT` instead of an empty string for consistency with the rest of the code. Consider changing it to `(sortField !== SortType.DEFAULT || reverse)`.